### PR TITLE
Some for loop optimizations

### PR DESCRIPTION
**Pull request by kranfix**
Many for converts unnecessarily from `Iterable` to `List` and, in some cases, make `map` operation instead of make it directly in the same `for`.

### CODE REVIEW

File: example/pubspec.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ description: A new Flutter project.
 version: 1.0.0+1
 
 environment:
-  sdk: ">=2.1.0 <3.0.0"
+  sdk: ">=2.3.0 <3.0.0"
```
Comment (diegoveloper) [inline example/pubspec.yaml]: we should allow use this package from 2.1.0 sdk version

File: lib/src/bezier_chart_widget.dart [hunk 2]
```diff
@@ -151,26 +151,24 @@ class BezierChartState extends State<BezierChart>
   List<double> _tempYValues;
 
   ///Refresh the position of the vertical/bubble
-  _refreshPosition(details) {
+  void _refreshPosition(details) {
     if (_animationController.status == AnimationStatus.completed &&
         _displayIndicator) {
-      _updatePosition(details);
+      return _updatePosition(details.globalPosition);
```
Comment (diegoveloper) [inline lib/src/bezier_chart_widget.dart]: why `return` if this is a `void` function and there is no more code after the `if` condition

File: pubspec.yaml [hunk 3]
```diff
@@ -9,12 +9,12 @@ homepage: https://aeyrium.com/
 repository: https://github.com/aeyrium/bezier-chart
 
 environment:
-  sdk: ">=2.1.0 <3.0.0"
+  sdk: ">=2.3.0 <3.0.0"
```
Comment (diegoveloper) [inline pubspec.yaml]: we should allow use this package from 2.1.0 sdk version

File: lib/src/bezier_chart_widget.dart (hunk 2 above)
Comment (kranfix) [reply lib/src/bezier_chart_widget.dart]: This avoids to use more callstack. Here I explain how it works but in Spanish: https://medium.com/@kranfix/optimizaci%C3%B3n-de-la-pila-de-llamadas-en-funciones-recursivas-796b3286ecd4

File: pubspec.yaml (hunk 3 above)
Comment (kranfix) [reply pubspec.yaml]: Ok, I'll downgrade the sdk version.

**Comment by diegoveloper** [comment]
There is an error on this file : `bezier_chart_widget.dart` line 649  , this method `intOrDouble` doesn't exist.
Try to remove this import also :

```
import 'package:flutter/rendering.dart' as prefix0;
```

**Comment by kranfix** [comment]
I don't remember to write prefix0. Maybe it was the IDE.
On the other hand, function name `intOrDouble` does not show what ir really does, then I renamed the function `formatAsIntOrDouble`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## [1.0.9]
+
+- Optimizing for loops removing unneessary convertion from `Iterable` to `List`.
```
Comment (diegoveloper) [inline CHANGELOG.md]: typo here

Comment (diegoveloper) [reply CHANGELOG.md]: don't worry I will fix this

**Comment by diegoveloper** [comment]
@kranfix thanks for the contribution

**Pull request merged**